src/scripts/inputManager.js

```diff
@@ -84,168 +89,77 @@ define(['playbackManager', 'focusManager', 'appRouter', 'dom', 'apphost'], funct
             }
         }
 
-        switch (name) {
-            case 'up':
-                focusManager.moveUp(sourceElement);
-                break;
-            case 'down':
-                focusManager.moveDown(sourceElement);
-                break;
-            case 'left':
-                focusManager.moveLeft(sourceElement);
-                break;
-            case 'right':
-                focusManager.moveRight(sourceElement);
-                break;
-            case 'home':
-                appRouter.goHome();
-                break;
-            case 'settings':
-                appRouter.showSettings();
-                break;
-            case 'back':
+        const keyActions = (command) => ({
+            'up': focusManager.moveUp(sourceElement),
+            'down': focusManager.moveDown(sourceElement),
+            'left': focusManager.moveLeft(sourceElement),
+            'right': focusManager.moveRight(sourceElement),
+            'home': appRouter.goHome(),
```

Comment:
This approach calls all of these functions.
Maybe something like `bind` will save us, but I was happy with `switch`. :p

Reply:
I'll investigate the function calls :)

Visually, I find this nicer. It also seems like this is more optimized in general, especially in cases where there are a lot of cases in the switch statement.

See here for more details: https://sonarcloud.io/organizations/mrtimscampi/rules?open=javascript%3AS1479&rule_key=javascript%3AS1479

Reply:
Fixed the potential issues by wrapping them in functions and adding `.call()` after the result is returned from the table.

It now works as expected. I also added error handling by sending a message to the debug channel when an unhandled event is sent.